test/python/pulse/test_instruction_schedule_map.py

```diff
@@ -205,3 +207,27 @@ def test_sequenced_parameterized_schedule(self):
         self.assertEqual(sched.instructions[0][-1].command.phase, 1)
         self.assertEqual(sched.instructions[1][-1].command.phase, 2)
         self.assertEqual(sched.instructions[2][-1].command.phase, 3)
+
+    def test_schedule_generator(self):
+        """Test schedule generator functionalty."""
+
+        x_test = 10
+        amp_test = 1.0
+
+        def test_func(x):
+            sched = Schedule()
+            sched += pulse_lib.constant(int(x), amp_test)(DriveChannel(0))
+            return sched
+
+        ref_sched = Schedule()
+        ref_sched += pulse_lib.constant(x_test, amp_test)(DriveChannel(0))
+
+        inst_map = InstructionScheduleMap()
+        inst_map.add('f', (0,), test_func)
+        self.assertEqual(inst_map.get('f', (0,), x_test), ref_sched)
+
+        self.assertIsInstance(inst_map.get_signature('f', (0,)), Signature)
```

Comment:
Can you add a test showing what this signature looks like for the user?

Reply:
I don't think I should test this, it could be Python version dependent? It's defined by the Python [inspect package](https://docs.python.org/3/library/inspect.html).